pinot-common/src/main/java/org/apache/pinot/common/config/DefaultClusterConfigChangeHandler.java

```diff
@@ -37,12 +40,12 @@
 public class DefaultClusterConfigChangeHandler implements ClusterConfigChangeListener, PinotClusterConfigProvider {
   private static final Logger LOGGER = LoggerFactory.getLogger(DefaultClusterConfigChangeHandler.class);
 
-  private volatile Map<String, String> _properties;
-  private final List<PinotClusterConfigChangeListener> _clusterConfigChangeListeners;
+  private volatile ImmutableMap<String, String> _properties;
+  private final CopyOnWriteArrayList<PinotClusterConfigChangeListener> _clusterConfigChangeListeners;
```

Comment:
Why is this change needed? Shouldn't the config change listeners just be registered at service startup before the Helix connection is established?